### ISSUE TITLE
SNS Create Topic fails when using tags when called more than once

### ISSUE THREAD

**Issue by Darth-Knoppix** [localstack/localstack#3928]
# Bug Report

[x] bug report
[ ] feature request

# Detailed description

When creating a topic for SNS with tags more than once the operation fails and is not idempotent. Attempting to create an SNS topic multiple times (same name) without specifying tags, works as expected. 

**Hypothesis:**
This only occurs with tags and seems to be caused by this line in [localstack.services.sns.sns_listener](https://github.com/localstack/localstack/blob/d142d415b6affd975025f7cf73b81b9450ff72fc/localstack/services/sns/sns_listener.py#L150) because `_extract_tags` attempts to compare two lists to see if the tags match.

## Expected behavior

When executing the following multiple times, a single topic should be created and subsequent executions will succeed as no tags have changed.

e.g. `awslocal sns create-topic --name coffee-orders --tags Key=a,Value=1 Key=b,Value=2`

**Original Python example:**

```py
client = boto3.client('sns', **aws_client_kwargs)
response = client.create_topic(
    Name='coffee-orders',
    Tags=[
        {"Key": "a", "Value": '1'},
        {"Key": "b", "Value": '2'},
    ]
)
```

**Results in:**
```
botocore.errorfactory.InvalidParameterException: An error occurred (InvalidParameter) when calling the CreateTopic operation: Topic already exists with different tags
```

## Actual behavior

The following error occurs:
`An error occurred (InvalidParameter) when calling the CreateTopic operation: Topic already exists with different tags`

# Steps to reproduce

## Command used to start LocalStack

`docker run --rm -p 4566:4566 -p 4571:4571 -e DEFAULT_REGION=eu-west-1 localstack/localstack-full`

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)

1. `awslocal sns create-topic --name coffee-orders --tags Key=a,Value=1 Key=b,Value=2`
2. `awslocal sns create-topic --name coffee-orders --tags Key=a,Value=1 Key=b,Value=2`

Running the command multiple times will result in the error.

**Comment by whummer** [comment]
Thanks for reporting @Darth-Knoppix . This should be fixed in #3930 - can you please pull the `latest` Docker image and give it another try? Please let us know if the problem persists. Thanks!

**Comment by Darth-Knoppix** [comment]
Thanks @whummer, this was a big blocker. I'll pull the latest and retest. Thanks again 👍
